examples/encoding/create_dash_multiperiod_from_keyframes_and_ad.py

```diff
@@ -0,0 +1,342 @@
+                        cloud_region=CloudRegion.AWS_EU_WEST_1)
+    encoding = bitmovin.encodings.Encoding.create(encoding).resource
+
+    # Generate keyframes
+    keyframes = []
+    for point in ad_insertion_points:
+        keyframe = Keyframe(time=point, segment_cut=True)
+        keyframe = bitmovin.encodings.Keyframe.create(object_=keyframe, encoding_id=encoding.id).resource
+        keyframes.append(keyframe)
+
+    content_resources = encode_file_to_fmp4(encoding=encoding,
+                                            input=https_input,
+                                            input_path=HTTPS_INPUT_PATH,
+                                            output=s3_output,
+                                            output_path=OUTPUT_BASE_PATH,
+                                            encoding_configs=encoding_configs)
+
+    # === Encode the ad ====
+    ad_resources = None
+    ad_encoding = None
+    if INSERT_PRE_ROLL or INSERT_MID_ROLLS or INSERT_POST_ROLL:
+        # Create an HTTPs input. This resource is then used as base URL for the input file.
+        ad_https_input = HTTPSInput(name='HTTPS input for ad', host=AD_HTTPS_INPUT_HOST)
+        ad_https_input = bitmovin.inputs.HTTPS.create(ad_https_input).resource
+
+        # Create an Encoding.
+        ad_encoding = Encoding(name="Multiperiod Dash - Ad",
+                               cloud_region=CloudRegion.AWS_EU_WEST_1)
+        ad_encoding = bitmovin.encodings.Encoding.create(ad_encoding).resource
+
+        ad_resources = encode_file_to_fmp4(encoding=ad_encoding,
+                                           input=ad_https_input,
+                                           input_path=AD_HTTPS_INPUT_PATH,
+                                           output=s3_output,
+                                           output_path=OUTPUT_BASE_PATH + "ad/",
+                                           encoding_configs=encoding_configs)
+
+    generate_multiperiod_dash_manifest(output=s3_output,
+                                       output_path=OUTPUT_BASE_PATH,
+                                       content_encoding_configs=content_resources,
+                                       keyframes=keyframes,
+                                       content_encoding=encoding,
+                                       ad_encoding=ad_encoding,
+                                       ad_encoding_configs=ad_resources)
+
+
+def encode_file_to_fmp4(encoding, input, input_path, output, output_path, encoding_configs):
+    video_input_stream = StreamInput(input_id=input.id,
+                                     input_path=input_path,
+                                     selection_mode=SelectionMode.AUTO)
+    audio_input_stream = StreamInput(input_id=input.id,
+                                     input_path=input_path,
+                                     selection_mode=SelectionMode.AUTO)
+
+    new_encoding_configs = copy.deepcopy(encoding_configs)
```

Comment:
are you sure that this works? can't find the copy import.

Reply:
You're right, I forgot to copy the import statement.
I've pushed the fix back into my fork - don't see the PR updating however for some reason